include/arch/arm/irq.h

```diff
@@ -136,6 +137,49 @@ static inline void arch_isr_direct_footer(int maybe_swap)
 	} \
 	static inline int name##_body(void)
 
+#if defined(CONFIG_DYNAMIC_DIRECT_INTERRUPTS)
+
+extern void z_arm_irq_direct_dynamic_dispatch_reschedule(void);
+extern void z_arm_irq_direct_dynamic_dispatch_no_reschedule(void);
+
+/**
+ * @brief Macro to register an ISR Dispatcher (with/without re-scheduling
```

Comment:
One thing that you are not documenting here is whether the user should use `IRQ_DIRECT_DECLARE` or just use a `void isr(void)` function instead.

Reply:
As additional info, direct interrupts are declared as `static inline int name##_body(void);`